src/eve/codegen.py

```diff
@@ -419,36 +419,44 @@ def render_template(self, **kwargs: Any) -> str:
 
 
 class TemplatedGenerator(NodeVisitor):
-    """A code generator visitor using :class:`TextTemplate` s.
+    """A code generator visitor using :class:`TextTemplate`.
 
-    The order followed to choose a `dump()` function for instances of
-    :class:`eve.Node` is the following:
+    The order followed to choose a `dump()` function for node values is the following:
 
-        1. A `self.visit_NODE_TYPE_NAME()` method where `NODE_TYPE_NAME`
-            matches `NODE_CLASS.__name__`, and `NODE_CLASS` is the
-            actual type of the node or any of its superclasses
-            following MRO order.
-        2. A `NODE_TYPE_NAME` class variable of type :class:`Template`,
-            where `NODE_TYPE_NAME` matches `NODE_CLASS.__name__`, and
-            `NODE_CLASS` is the actual type of the node or any of its
-            superclasses following MRO order.
+        1. A ``self.visit_NODE_CLASS_NAME()`` method where `NODE_CLASS_NAME`
+           matches ``type(node).__name__``.
+        2. A ``self.visit_NODE_BASE_CLASS_NAME()`` method where
+           `NODE_BASE_CLASS_NAME` matches ``base.__name__``, and `base` is
+           one of the node base classes (evaluated following the order
+           given in ``type(node).__mro__``).
```

Comment:
Maybe add a comment that a typical pattern is to call `generic_visit()` in the end to process templates of the same name.

Reply:
Done